app/src/main/java/com/vitorpamplona/amethyst/ui/components/Robohash.kt

```diff
@@ -0,0 +1,299 @@
+package com.vitorpamplona.amethyst.ui.components
+
+import android.content.Context
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.toArgb
+import coil.request.ImageRequest
+import java.nio.ByteBuffer
+import java.security.MessageDigest
+
+private fun toHex(color: Color): String {
+    val argb = color.toArgb()
+    val rgb = argb and 0x00FFFFFF // Mask out the alpha channel
+    return String.format("#%06X", rgb)
+}
+
+private val sha256: MessageDigest = MessageDigest.getInstance("SHA-256")
+
+private fun byteMod(byte: Byte, modulo: Int): Int {
+    val ub = byte.toUByte().toInt()
+    return ub % modulo
+}
+
+private fun bytesToRGB(b1: Byte, b2: Byte, b3: Byte): Color {
+    return Color(b1.toUByte().toInt(), b2.toUByte().toInt(), b3.toUByte().toInt())
+}
+
+private fun svgString(msg: String): String {
+    val hash = sha256.digest(msg.toByteArray())
+    val hashHex = hash.joinToString(separator = "") { b -> "%02x".format(b) }
+    val bgColor1 = bytesToRGB(hash[0], hash[1], hash[2])
+    val bgColor2 = bytesToRGB(hash[3], hash[4], hash[5])
+    val fgColor = bytesToRGB(hash[6], hash[7], hash[8])
+    val bgIndex = byteMod(hash[9], 8)
+    val bodyIndex = byteMod(hash[10], 10)
+    val faceIndex = byteMod(hash[11], 10)
+    val eyesIndex = byteMod(hash[12], 10)
+    val mouthIndex = byteMod(hash[13], 10)
+    val accIndex = byteMod(hash[14], 10)
+    val background = backgrounds[bgIndex]
+    val body = bodies[bodyIndex]
+    val face = faces[faceIndex]
+    val eye = eyes[eyesIndex]
+    val mouth = mouths[mouthIndex]
+    val accessory = accessories[accIndex]

```

Comment:
Hash a message and pick some values from the bytes (not sure if there is a preferred way to do this, I just did it arbitrarily...)